CMakeLists.txt

```
@@ -119,6 +119,13 @@ target_include_directories(lsp PUBLIC
 	${CMAKE_CURRENT_SOURCE_DIR}
 	${LSP_GENERATED_FILES_DIR}
 )
+install(TARGETS lsp LIBRARY)
+include(GNUInstallDirs)
+install(FILES
+	${LSP_GENERATED_FILES_DIR}/lsp/types.h
+	${LSP_GENERATED_FILES_DIR}/lsp/messages.h
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lsp)
+install(DIRECTORY lsp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

You should only install the headers not the sources.

```suggestion
install(TARGETS lsp ARCHIVE LIBRARY)
include(GNUInstallDirs)
install(
	FILES ${LSP_GENERATED_HEADERS}
	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lsp
)
install(
	DIRECTORY ${LSP_SRC_DIR}
	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lsp
	FILES_MATCHING PATTERN "*.h"
)
```

---

> FILES_MATCHING PATTERN "*.h"

You forget `#include "messagehandler.inl"`